src/rosecommon/include/crosewriter.h

```diff
@@ -90,14 +108,14 @@ class CRoseReader {
     public:
         CRoseReader(const uint8_t *buffer, uint16_t size) : m_current(buffer), m_buffer(buffer), m_size(size) {}
 
-        bool get_uint8_t(uint8_t& data) { return read(data); }
-        bool get_int8_t(int8_t& data) { return read(data); }
-        bool get_uint16_t(uint16_t& data) { return read(data); }
-        bool get_int16_t(int16_t& data) { return read(data); }
-        bool get_uint32_t(uint32_t& data) { return read(data); }
-        bool get_int32_t(int32_t& data) { return read(data); }
-        bool get_uint64_t(uint64_t& data) { return read(data); }
-        bool get_int64_t(int64_t& data) { return read(data); }
+        bool get_uint8_t(uint8_t& data, size_t bits=sizeof(uint8_t)) { return read(data, bits); }
```

Comment:
Naming is confusing, the optional parameter is called `bits` but it actually is the number of bytes...

Reply:
thank you for saying this. I was thinking the same thing. I was thinking that I was reading this incorrectly.